docs/source/1.0/guides/migrationg-idl-1-to-2.rst

```diff
@@ -0,0 +1,292 @@
+with 2.0, but instead are refactoring steps that can be taken to simplify a
+your model.
+
+
+Move operation inputs and outputs inline
+----------------------------------------
+
+The structures that define operation inputs and outputs very often use
+boilerplate names and for readability are usually placed close to their parent
+operation shapes to improve readability of the model. Smithy 2.0 introduced
+:ref:`inline input and output <idl-inline-input-output>`, which allows you
+to define those shapes as part of the definition of the operation rather than
+separately. This improves readability and reduces the amount of boilerplate
+needed to model an operation. For example, the following model:
+
+.. code-block:: smithy
+
+    $version: "1.0"
+
+    namespace smithy.example
+
+    operation PutUser {
+        input: PutUserInput,
+        output: PutUserOutput
+    }
+
+    @input
+    structure PutUserInput {
+        email: String,
+        id: String,
+        username: String,
+        description: String
+    }
+
+    @output
+    structure PutUserOutput {}
+
+can be updated to:
+
+.. code-block::
+
+    $version: "2.0"
+
+    namespace smithy.example
+
+    operation PutUser {
+        input := {
+            email: String
+            id: String
+            username: String
+            description: String
+        },
+        output := {}
+    }
+
+.. seealso::
+
+    the :ref:`inline input / output <idl-inline-input-output>` section of the
+    spec for more details.
+
+
+Abstract shared shape configuration with mixins
+-----------------------------------------------
+
+Models often have several shapes that refer to the same sets of members, or
+which share a set of trait configurations. For example, resource instance
+operations all require that the resource's identifiers be present. With
+:ref:`mixins`, it is easy to simply share these member definitions without
+having to copy and paste them. The following model:
+
+.. code-block:: smithy
+
+    $version: "1.0"
+
+    namespace smithy.example
+
+    resource User {
+        identifiers: {
+            email: String,
+            id: String,
+            username: String
+        },
+        read: GetUser
+    }
+
+    operation GetUser {
+        input: GetUserInput,
+        output: GetUserOutput
+    }
+
+    @input
+    structure GetUserInput {
+        email: String,
+        id: String,
+        username: String
+    }
+
+    @output
+    structure GetUserOutput {
+        email: String,
+        id: String,
+        username: String,
+        description: String
+    }
+
+Can be updated to:
+
+.. code-block:: smithy
+
+    $version: "2.0"
+
+    namespace smithy.example
+
+    resource User {
+        identifiers: {
+            email: String
+            id: String
+            username: String
+        },
+        read: GetUser
+    }
+
+    @mixin
+    structure UserIdentifiers {
+        email: String
```

Comment:
These members all need to be required... Might be nice to just trim down the example to one identifier too.

Reply:
How about two? I want there to be more than one to emphasize how annoying it is to copy and paste.